### PR TITLE
Inline some encoding and decoding methods.

### PR DESCRIPTION
**Pull request by nnethercote**
This is a small performance win.

r? @Centril

### CODE REVIEW

**Comment by nnethercote** [comment]
Local `check` results:
```
clap-rs-check
        avg: -0.5%      min: -1.1%      max: 0.0%
ucd-check
        avg: -0.3%      min: -0.7%      max: -0.0%
html5ever-check
        avg: -0.2%      min: -0.3%      max: -0.0%
wg-grammar-check
        avg: -0.1%      min: -0.3%      max: -0.0%
tuple-stress-check
        avg: -0.1%      min: -0.3%      max: -0.0%
script-servo-check
        avg: -0.1%      min: -0.2%      max: 0.0%
keccak-check
        avg: -0.1%      min: -0.2%      max: 0.0%
coercions-check
        avg: -0.0%?     min: -0.2%?     max: 0.1%?
cranelift-codegen-check
        avg: -0.1%      min: -0.2%      max: -0.0%
ctfe-stress-4-check
        avg: 0.1%?      min: -0.0%?     max: 0.2%?
wf-projection-stress-65510-che...
        avg: -0.1%      min: -0.2%      max: -0.0%
await-call-tree-check
        avg: -0.1%      min: -0.2%      max: -0.1%
cargo-check
        avg: -0.1%      min: -0.1%      max: -0.0%
regression-31157-check
        avg: -0.1%      min: -0.1%      max: -0.0%
regex-check
        avg: -0.1%      min: -0.1%      max: -0.0%
piston-image-check
        avg: -0.1%      min: -0.1%      max: -0.0%
inflate-check
        avg: -0.0%      min: -0.1%      max: -0.0%
webrender-check
        avg: -0.1%      min: -0.1%      max: -0.0%
helloworld-check
        avg: -0.1%      min: -0.1%      max: -0.1%
serde-serde_derive-check
        avg: -0.1%      min: -0.1%      max: -0.0%
unify-linearly-check
        avg: -0.1%      min: -0.1%      max: -0.0%
unused-warnings-check
        avg: -0.1%      min: -0.1%      max: -0.0%
syn-check
        avg: -0.1%      min: -0.1%      max: -0.0%
issue-46449-check
        avg: -0.1%      min: -0.1%      max: -0.1%
deeply-nested-check
        avg: -0.1%      min: -0.1%      max: -0.0%
futures-check
        avg: -0.1%      min: -0.1%      max: 0.0%
hyper-2-check
        avg: -0.0%      min: -0.1%      max: -0.0%
style-servo-check
        avg: -0.0%      min: -0.1%      max: 0.0%
trait-stress-check
        avg: -0.0%      min: -0.1%      max: -0.0%
serde-check
        avg: -0.0%      min: -0.1%      max: 0.0%
encoding-check
        avg: -0.1%      min: -0.1%      max: -0.0%
ripgrep-check
        avg: -0.1%      min: -0.1%      max: -0.0%
deep-vector-check
        avg: -0.0%      min: -0.1%      max: -0.0%
webrender-wrench-check
        avg: -0.1%      min: -0.1%      max: -0.0%
tokio-webpush-simple-check
        avg: -0.0%      min: -0.1%      max: -0.0%
unicode_normalization-check
        avg: -0.0%      min: -0.1%      max: 0.1%
packed-simd-check
        avg: -0.0%      min: -0.1%      max: -0.0%
token-stream-stress-check
        avg: -0.0%      min: -0.0%      max: -0.0%
```
A small but clear win, not bad for such a simple change.

**Comment by Centril** [comment]
r=me with perf improved

**Comment by nnethercote** [comment]
It's a miniscule win, but a win nonetheless.

@bors r=Centril

File: src/librustc/ty/query/on_disk_cache.rs [hunk 1]
```diff
@@ -943,6 +943,7 @@ where
 
 macro_rules! encoder_methods {
     ($($name:ident($ty:ty);)*) => {
+        #[inline]
         $(fn $name(&mut self, value: $ty) -> Result<(), Self::Error> {
```
Comment (ecstatic-morse) [inline src/librustc/ty/query/on_disk_cache.rs:947]: The `#[inline]` here is only applied to the first method. Always putting `$(` on its own line would have avoided this. cc rust-lang/rustfmt#8

**Pull request merged**